docs/src/stories/components/Layout/Stack.stories.jsx

```diff
@@ -0,0 +1,464 @@
+import React from 'react'
+import clsx from 'clsx'
+
+export default {
+  title: 'Components/Layout/Stack',
+  excludeStories: ['StackTemplate'],
+  argTypes: {
+
+    // Debug
+
+    _debug: {
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'Debug'
+      },
+    },
+
+    _height: {
+      control: {
+        type: 'number',
+      },
+      table: {
+        category: 'Debug'
+      },
+    },
+
+    _width: {
+      control: {
+        type: 'number',
+      },
+      table: {
+        category: 'Debug'
+      },
+    },
+
+    // Direction
+
+    direction: {
+      options: ['inline', 'block'],
+      control: {
+        type: 'inline-radio',
+      },
+      description: 'Sets how elements inside `Stack` are placed, either horizontally (`inline`) or vertically (`block`). This property follows the writing mode.',
+      table: {
+        category: 'Properties',
+        defaultValue: {
+          summary: 'block',
+        }
+      }
+    },
+    narrow_direction: {
+      options: ['inherit', 'inline', 'block'],
+      control: {
+        type: 'inline-radio',
+      },
+      description: 'Override `direction` on narrow viewports',
+      table: {
+        category: 'Narrow viewport properties',
+        defaultValue: {
+          summary: 'inherit'
+        }
+      },
+    },
+
+    // Gap
+
+    gap: {
+      options: ['none', 'condensed', 'normal', 'spacious', 'custom'],
```

Comment:
I don't think we should start with allowing custom gap values. We could add it later if we actually need it.

Reply:
@mperrotti we actually need it already while we don't have a specialized component for handling a horizontal stack of controls. I've included that in the PR description 🙇 

> **Note on using inline Stack for buttons**
>
> In the future we may have specialized stacking components for controls using the [UI control stack tokens](https://primer.style/primitives/spacing#ui-control-stack).
>
> In the meantime, we can use these tokens with a custom gap:
> ```html
> <Stack direction="inline" spread="end" gap="var(--primer-controlStack-medium-gap-auto)">
>   <button>Cancel</button>
>   <button>Save</button>
> </Stack>
> ```

Reply:
@mperrotti the more I think about it, the more I agree with you 🌞 . Let's not add support for custom gap in PVC and PRC implementations to start. For buttons, we can provide specialized components that handle that with the right design tokens 🙇 🙇

Reply:
I've hidden the custom gap options from Storybook for now as well.